haystack/components/generators/openai.py

```diff
@@ -189,6 +193,8 @@ def run(
         for each response.
         """
         message = ChatMessage.from_user(prompt)
+        if system_prompt is not None:
+            self.system_prompt = system_prompt
```

Comment:
I'd do this slightly differently, I don't want the `self` version to be updated, just optionally overwritten at run time. 

```suggestion
        final_system_prompt = self.system_prompt
        if system_prompt is not None:
           final_system_prompt = system_prompt
```
and then update the below code to use `final_system_prompt` instead of `self.system_prompt`

Reply:
Feel free to find a better name than `final_system_prompt` :D

Reply:
ok, I see the use case 👍🏽